proxy/tests/telemetry.rs

```diff
@@ -1090,4 +1090,35 @@ mod transport {
         drop(tcp_client);
         assert_contains!(metrics.get("/metrics"), &expected);
     }
+
+    #[test]
+    #[cfg_attr(not(feature = "flaky_tests"), ignore)]
+    fn outbound_tcp_connections_open() {
+        let _ = env_logger::try_init();
+        let TcpFixture { client, metrics, proxy: _proxy } =
+            TcpFixture::outbound();
+
+        let msg1 = "custom tcp hello";
+        let msg2 = "custom tcp bye";
+
+        let tcp_client = client.connect();
+
+        tcp_client.write(msg1);
+        assert_eq!(tcp_client.read(), msg2.as_bytes());
+        assert_contains!(metrics.get("/metrics"),
+            "tcp_connections_open{direction=\"outbound\"} 2");
```

Comment:
why are there 2 outboudn connections here? I only see one call to `client.connect` before this?

Reply:
Is this counting the server and client connections independently? That seems surprising -- since they're really just a single logical connection that's being managed by the proxy... if that is what's going on, we may need to differentiate them with labels. In the shorter term, it seems best to only measure this on the accept-side and not count client connections...

Reply:
Okay, that sounds good. 

I also thought that there should be a label for differentiating between opened and accepted connections; I think I misinterpreted one of your comments on #742 as saying that this gauge should track server and client connections separately.  

I think adding an additional label shouldn't be much more work, but for now I'll just make this track the accepted connections only?

Reply:
that sounds right.

Another question: does this count connections that are determined to be HTTP as well? including an http route in the test would be helpful

Reply:
Will do --- I thought I did, but it appears I was wrong

Reply:
Fixed in 83e9fc53a10ebf081f0468aad2e304a2cd5fe21d